tests/unit/test-util/test.submit-addon.js

```diff
@@ -0,0 +1,520 @@
+/* @flow */
+import { promises as fsPromises, readFileSync } from 'fs';
+import path from 'path';
+
+import { fs } from 'mz';
+import { assert, expect } from 'chai';
+import { afterEach, before, beforeEach, describe, it } from 'mocha';
+import * as sinon from 'sinon';
+import nock from 'nock';
```

Comment:
During this review pass I took a look into how we may get rid of `nock` as an additional dev dependency, and given that in the submit-addon.js module we are already wrapping `fileFromSync` in a `Client` method for the purpose of mocking it in tests, it looks that we can use the same approach with `node-fetch` and be able to fully replace `nock` with some more `sinon.stub` calls.

Given that I was looking the submit-addon.js module in a bit more detail during this pass, I also thought to give that a try, the result seems to be nice enough and so I'd like us to go for that:
- https://github.com/rpl/web-ext/commit/c192cef1454c11f49a297758e3bad377152119a5

The commit linked right above includes all the changes I've applied locally on top of the current version of this PR.

There is only one small addition to submit-addon.js (just the new method and one node-fetch call replaced with a call to the new class instance method) and a couple of helpers defined inside test.submit-addon.js (a `JSONResponse` helper class and a `mockNodeFetch` test helper to reduce the verbosity of the common sinon.stub calls needed to mock the requests in a way that too much more verbose they how `nock` used to be, with these two helpers the conversion is almost 1-1 with `nock`).

Besides that the commit linked above includes an additional small change on the `waitRetry` method, I'll comment about that in a separate review comment, nearby the code being tweaked.

Reply:
I quite liked `nock` - not too big, focused, hooked into everywhere, and didn't require lots of config each time. It wouldn't have been high up on my list of dependencies to rewrite the code into the repo but 🤷.

Reply:
I wasn't openly against `nock`, but not having to introduce a new dependencies is still nice, less transitive dependencies that may be outdated and get a new security advisory to deal with.

If replacing it with `sinon.stub` was going to be much more verbose or too far from how nice `nock` based mocking was, than I would have been ok to add `nock`, but honestly the replacement result looked nice enough (and we may be also avoiding some magic that `nock` may be using internally, which maybe was why the same test written with `nock` was not catching the issue triggered by the `waitForRetry` method).